### ISSUE TITLE
PID & VID for CH PRO peddels

### ISSUE THREAD

**Issue by GeekAnnoyed** [redmcg/wine#10]
More of a Question
how do I find my pid and vid in https://github.com/redmcg/wine/wiki/Peripherals

my CH PRO peddles are not being detected so I was hoping this would fix them maybe.

**Comment by redmcg** [comment]
Assuming it's connected via USB - then running `lsusb` should tell you. Here's the output for my controller:
`Bus 002 Device 004: ID 046d:c21d Logitech, Inc. F310 Gamepad [XInput Mode]`

This bit: `046d:c21d`; is the VID and PID (VID = 0x046d and PID = 0xc21d).

Although I did find this via Google:
```
Device Name: CH PRO PEDALS USB 
Vendor ID (VID): 0x068E
Product ID (PID): 0x00F2
```

But running `lsusb` will confirm.

**Comment by GeekAnnoyed** [comment]
Oh, I was really hoping they were different for windows.

As lsusb is what I was doing and no joy in finding a reg entry.
I did notice when I ran the wine control utility i had a /Dev/input/js0 not a event one

Sorry for my brevity this is all being done mobile and from memory 😃.

**Comment by redmcg** [comment]
Have a look at my comment here for some things to try:
https://github.com/redmcg/wine/issues/8#issuecomment-468409644

Especially this bit:
`$ /tmp/proton_user/run control joy.cpl`

If under the 'Test Joystick' tab you see your device with the (js) option; and it works - then you should be able to get it going in Elite.

The (js) refers to devices with a `/dev/input/js` entry.

![joycpl](https://user-images.githubusercontent.com/8346438/54722823-af160680-4bb9-11e9-9c3c-72a58cc504a0.png)

**Comment by GeekAnnoyed** [comment]
HI there
it wasn't displayed in the control panel.
after running "/tmp/proton_user/run control joy.cpl"
![2019-03-21_13-01](https://user-images.githubusercontent.com/822758/54753688-93baf380-4bd9-11e9-8ccf-e5510b35a6ad.png)

here is the log from steam
[steam-359320.tar.gz](https://github.com/redmcg/wine/files/2992089/steam-359320.tar.gz)

edit:
as a side note I just tried Flight Sim World on proton and the pedals aren't detected either.

**Comment by redmcg** [comment]
Sorry - I forgot Proton doesn't use the (js) driver - it uses SDL2; which I think uses evdev (and not js).

You might want to try this:
http://thegnuflyer.blogspot.com/2014/12/ch-pro-pedals-and-linux-x-plane.html?m=1

And also try joy.cpl in standard wine:
`wine64 control joy.cpl`

**Comment by GeekAnnoyed** [comment]
it's perfectly okay :D 
the gnuflyer is my old abandoned blog (must delete it)
do you have any pedals if so what ones? it was time for a upgrade anyway

**Comment by redmcg** [comment]
Haha - that's embarrassing.

I only use a controller.

I'm curious if your peddles show up in standard wine though?

**Comment by GeekAnnoyed** [comment]
it was a https://xkcd.com/979/ moment for me too till I realised 
they do show up but only as a /dev/input/js0

oh well looks like i'm doing some research

**Comment by redmcg** [comment]
The game runs with a recent version of Wine:
https://github.com/redmcg/wine/wiki/Wine

The pedals should work if they show up in joy.cpl.

For Proton it means getting them to work with SDL2 - which I think means getting them to show up in evdev (which they should if it's hid).

**Comment by redmcg** [comment]
You might want to try creating a file:
`/etc/﻿﻿﻿udev/rules.d/51-input-joystick.rules`

consisting of:

```
# "CH PRODUCTS CH PRO PEDALS USB "
SUBSYSTEM=="input", ENV{ID_VENDOR_ID}=="068e", ENV{ID_MODEL_ID}=="00f2", ENV{ID_INPUT_ACCELEROMETER}="0"
KERNEL=="event*", ENV{ID_VENDOR_ID}=="068e", ENV{ID_MODEL_ID}=="00f2﻿", ENV{ID_INPUT_JOYSTICK}="1"
```

Source:
https://forums.x-plane.org/index.php?/forums/topic/72733-writing-and-debugging-udev-rules/&do=findComment&comment=841134

**Comment by GeekAnnoyed** [comment]
well, good news 
tl;dr

for other users, I found a file on the X-Plane forums
cat /etc/udev/rules.d/30-x-plane.rules

```
# Saitek Pro Flight Cessna Rudder Pedals
SUBSYSTEMS=="input", ATTRS{name}=="Mad Catz Pro Flight Cessna Rudder Pedals", ENV{ID_INPUT_JOYSTICK}="1"

# Saitek Pro Flight Cessna Trim Wheel
SUBSYSTEMS=="input", ATTRS{name}=="Saitek Pro Flight Cessna Trim Wheel", ENV{ID_INPUT_JOYSTICK}="1"

# Saitek Pro Flight Rudder Pedals
SUBSYSTEMS=="input", ATTRS{name}=="Saitek Saitek Pro Flight Rudder Pedals", ENV{ID_INPUT_JOYSTICK}="1"

# Saitek Pro Flight Combat Rudder Pedals
SUBSYSTEMS=="input", ATTRS{name}=="Saitek Saitek Pro Flight Combat", ENV{ID_INPUT_JOYSTICK}="1"

# CH PRO PEDALS USB
SUBSYSTEMS=="input", ATTRS{name}=="CH PRODUCTS CH PRO PEDALS USB", ENV{ID_INPUT_JOYSTICK}="1"

```
 and it works!!! :dancing_men: :dancing_women: 


also I thought you might like to Know I'm AMD GPU user and I've had zero issues related to that :+1: 


Thank you so much for your time and google skills

**Comment by redmcg** [comment]
Fantastic! That is good news.

I'll close this issue - but do you mind updating the [AMD](https://github.com/redmcg/wine/wiki/AMD) section with the instructions for your distro? You probably noticed it's a little bare.

Also feel free to also update the [Peripherals](https://github.com/redmcg/wine/wiki/Peripherals) page with the above.

No problem if you're not interested - I just wanted to make sure you knew you were welcome to update them.
